packages/ra-supabase-ui-materialui/src/guessers/ListGuesser.tsx

```diff
@@ -0,0 +1,179 @@
+        queryOptions,
+        resource,
+        sort,
+        storeKey,
+        ...rest
+    } = props;
+    return (
+        <ListBase
+            debounce={debounce}
+            disableAuthentication={disableAuthentication}
+            disableSyncWithLocation={disableSyncWithLocation}
+            exporter={exporter}
+            filter={filter}
+            filterDefaultValues={filterDefaultValues}
+            perPage={perPage}
+            queryOptions={queryOptions}
+            resource={resource}
+            sort={sort}
+            storeKey={storeKey}
+        >
+            <ListGuesserView {...rest} />
+        </ListBase>
+    );
+};
+
+export const ListGuesserView = (
+    props: ListViewProps & {
+        enableLog?: boolean;
+    }
+) => {
+    const { data: schema, error, isPending } = useAPISchema();
+    const resource = useResourceContext();
+    const [child, setChild] = React.useState<ReactNode>(null);
+    const [filters, setFilters] = React.useState<
+        React.ReactElement[] | undefined
+    >(undefined);
+    if (!resource) {
+        throw new Error('ListGuesser must be used withing a ResourceContext');
+    }
+    const { enableLog = process.env.NODE_ENV === 'development', ...rest } =
+        props;
+
+    React.useEffect(() => {
+        if (isPending || error) {
+            return;
+        }
+        const resourceDefinition = schema.definitions?.[resource];
+        if (!resourceDefinition || !resourceDefinition.properties) {
+            throw new Error(
+                `The resource ${resource} is not defined in the API schema`
+            );
+        }
+        if (!resourceDefinition || !resourceDefinition.properties) {
+            return;
+        }
+        const inferredFields = Object.keys(resourceDefinition.properties).map(
+            (source: string) =>
+                inferElementFromType({
+                    name: source,
+                    types: listFieldTypes,
+                    description:
+                        resourceDefinition.properties![source].description,
+                    format: resourceDefinition.properties![source].format,
+                    type: (resourceDefinition.properties &&
+                    resourceDefinition.properties[source] &&
+                    typeof resourceDefinition.properties[source].type ===
+                        'string'
+                        ? resourceDefinition.properties![source].type
+                        : 'string') as string,
+                })
+        );
+        const inferredTable = new InferredElement(
+            listFieldTypes.table,
+            null,
+            inferredFields
+        );
+        setChild(inferredTable.getElement());
+
+        const rowFilters =
+            schema!
+                .paths![`/${resource}`].get!.parameters?.filter(obj =>
+                    obj['$ref'].includes('rowFilter')
+                )
+                .map(obj => obj['$ref'].split('.').pop()) ?? [];
```

Comment:
Do we want to have a filter on the `id` column?

Reply:
yes